### ISSUE TITLE
Library Request: keras-tcn

### ISSUE THREAD

**Issue by AlexCatarino** [QuantConnect/Lean#7810]
[Keras Temporal Convolutional Network](https://pypi.org/project/keras-tcn/#keras-tcn) 

Why Temporal Convolutional Network instead of LSTM/GRU?
- TCNs exhibit longer memory than recurrent architectures with the same capacity.
- Performs better than LSTM/GRU on a vast range of tasks (Seq. MNIST, Adding Problem, Copy Memory, Word-level PTB...).
- Parallelism (convolutional layers), flexible receptive field size (possible to specify how far the model can see), stable gradients (backpropagation through time, vanishing gradients)...

#### Checklist
- [X] I have completely filled out this template
- [X] I have confirmed that this issue exists on the current `master` branch
- [X] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)
